### DISCUSSION TITLE
Included Validation

### DISCUSSION THREAD

**Discussion by nam-co** [alpinejs/alpine#1072] [3. Feature ideas]
Hi Caleb, thanks for the hard work in alpine Js, just an idea, it could be great if there was a way to validate fields inside alpine, something like this maybe, nobody would bother about a few kb more
`
<span v-show="!email.validate">check your email</span>
<input x-model="email" x-on:blur.validate="email,max:50"> OR <input x-model="email" x-validate.debounce="email,max:50">
`
hope you'll consider it, have a great day

**Comment by philwolstenholme** [comment]
@nam-co I can't imagine Alpine adding built-in form validation but you might be interested in using a form validation library called Iodine alongside Alpine, for example like the approach shown in this CSS Tricks article: https://css-tricks.com/lightweight-form-validation-with-alpine-js-and-iodine-js

**Reply by nam-co** [reply]
Thanks @philwolstenholme , I saw something similar before posting but it seems like a lot of extra code, kind of ruins the simplicity of alpine, seems better going the vue + vee-validate way, but since alpine is used for a lot of simple show/update/push stuff maybe @calebporzio will consider it, a simple basic implementation
